eth/tracers/logger/logger.go

```diff
@@ -30,6 +30,7 @@ import (
 	"github.com/ethereum/go-ethereum/common/math"
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/eth/tracers/directory"
```

Comment:
So I fixed this potential import cycle from the core package. The reason it was happening is that `eth/tracers/directory` was importing to use `core.BlockchainLogger` for the LiveDirectory. I moved the live directory to a subpackage `eth/tracers/directory/live` so that now `eth/tracers/logger` can be imported from the core tests without a problem. It's a bit meh because the subpackage `directory/live` is literally 15 LoC but yeah.

Tho it wasn't working exactly out-of-the box. I had added a sanity check in blockchain to error out if the tracer passed in via vm.Config was not a live tracer (since blockchain invokes those events). Now I changed this behavior to emit a warning and proceed without setting the bc logger.